test/test_cuda.py

```diff
@@ -2118,16 +2119,43 @@ def test_cuda_memory_leak_detection_propagates_errors(self):
                 y = torch.randn(2, 1, device='cuda')
                 z = x + y
 
+    def _compare_trilu_indices(self, row, col, offset=0, dtype=torch.long):
+        if row == 0 or col == 0:
+            # have to handle this separately as tril and triu does not take
+            # empty matrix as input
+            self.assertEqual(
+                torch.empty(0, 2, dtype=dtype, device='cuda').transpose(0, 1),
+                torch.tril_indices(row, col, offset, dtype=dtype, device='cuda'))
+            self.assertEqual(
+                torch.empty(0, 2, dtype=dtype, device='cuda').transpose(0, 1),
+                torch.triu_indices(row, col, offset, dtype=dtype, device='cuda'))
+
+        else:
+            # explicitly convert 'cpu' tensor to 'cuda' to avoid a bug in tril
+            # and triu cuda kernel (see #15226)
+            x = torch.ones(row, col, dtype=dtype, device='cpu') \
+                     .tril(offset).nonzero().transpose(0, 1).cuda()
```

Comment:
To help with the deduplication between this and the _compare_trilu_indices in test_torch.py, you can do
```
x = torch.ones(row, col, dtype=dtype, device='cpu') \
                     .tril(offset).nonzero().transpose(0, 1).to(device)
```
here, assuming we pass in a 'device' for _compare_trilu_indices